src/users/blocks.repository.ts

```diff
@@ -0,0 +1,34 @@
+import { DataSource, Repository } from 'typeorm';
+import { Block } from './entities/block.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import { DEFAULT_PAGE_SIZE } from '../common/constants';
+import { BlockUserInfoDto } from './dto/block-user-info.dto';
+import { DBQueryErrorException } from '../common/exception/custom-exception';
+
+export class BlocksRepository extends Repository<Block> {
+	constructor(@InjectRepository(Block) private dataSource: DataSource) {
+		super(Block, dataSource.manager);
+	}
+
+	async findBlockUsers(
+		userId: string,
+		page: number,
+	): Promise<BlockUserInfoDto[]> {
+		const users = this.dataSource
+			.query(
+				`
+			SELECT u.id, u.nickname, u.avatar, u.status
+			FROM Block b
+			JOIN "user" u
+			ON u.id = b."toUserId" 
+			WHERE b."fromUserId" = $1 AND b."deletedAt" IS NULL // QueryFailedError: operator does not exist: uuid = character varying
+			LIMIT $2 OFFSET $3;
```

Comment:
```suggestion
	async findBlockUsers(
		userId: UUID,
		page: number,
	): Promise<BlockUserInfoDto[]> {
		const users = this.dataSource
			.query(
				`
			SELECT u.id, u.nickname, u.avatar, u.status
			FROM Block b
			JOIN "user" u
			ON u.id::uuid = b."toUserId"::uuid
			WHERE b."fromUserId"::uuid = ($1)::uuid AND b."deletedAt" IS NULL
			LIMIT $2 OFFSET $3;
```
이런식으로 cast해주면 됩니다! 그렇지만 정말 이 방법밖에 없나 고민중...

Reply:
@tomatozil 천천히 확인해주삼 ~